### PR TITLE
Use nyc for coverage, update mochify and browserify

### PR DESCRIPTION
**Pull request by mantoni**
#### Purpose (TL;DR)

As discussed in #1948, we want to replace mochify-istanbul since it's basically unmaintained and has outdated dependencies.

 #### Solution

This change upgrades Mochify to v6 which comes with `nyc` support, generating the coverage report with up-to-date istanbul instrumenter libraries.

More dependency updates:

- Mochify 6 uses Browserify 16 now, so the dependency was also updated here.
- Mochify 6 uses Puppeteer 1.10 now,, cause two Chromium downloads here, so this was also aligned.
- The obsolete mochify-istanbul dependency was removed.

 #### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm run test-coverage`

 #### Checklist for author

- [x] `npm run lint` passes
- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).
- [x] `npm test` passes – ~~__NOTE:__ This currently fails due to `npm run test-webworker` failing.~~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "test-node": "mocha --recursive -R dot \"test/**/*-test.js\"",
     "test-dev": "npm run test-node -- --watch -R min",
     "test-headless": "mochify --recursive -R dot --grep WebWorker --invert --plugin [ proxyquire-universal ] \"test/**/*-test.js\"",
-    "test-coverage": "mochify --recursive -R dot --grep WebWorker --invert --plugin [ proxyquire-universal ] --plugin [ mochify-istanbul --exclude '**/test/**' --report text --report lcovonly --dir ./coverage ] \"test/**/*-test.js\"",
+    "test-coverage": "nyc npm run test-headless -- --transform [ babelify --ignore [ test ] --plugins [ babel-plugin-istanbul ] ]",
```
Comment (mroderick) [inline package.json]: At least the line is shorter :P

File: package.json [hunk 2]
```diff
@@ -58,8 +66,11 @@
     "type-detect": "^4.0.8"
   },
   "devDependencies": {
+    "@babel/core": "^7.1.6",
```
Comment (mroderick) [inline package.json]: Ugh! How long do you think it'll be before people start suggesting that we rewrite the entire codebase in stage-0 ES?

Comment (mantoni) [reply package.json]: We've been fighting it successfully so far 😉

Comment (mroderick) [reply package.json]: "But you're already loading babel..."

Comment (mantoni) [reply package.json]: It's actually not such a bad idea to verify that sinon works fine when babelified. However, the one strong argument that we can always pull is that Sinon works very low level and we need complete control over the code that runs. Therefore, the codebase is written soley with language features that are supported by the targeted runtimes natively.

**Comment by mroderick** [comment]
> So I double checked that the webworker is able to load the file `pgk/sinon.js`. There seems to be an error with the script execution within the webworker. Unfortunately, we don't see that error. I'd be very thankful for pointers on how to debug this.

Perhaps @fatso83 can give a pointer, since they spent some time in this part of the codebase (I think 🤔)

**Comment by mantoni** [comment]
Final touches:

- Upgraded Mocha to v5.2, so now Node and Browser tests use the same Mocha version everywhere.
- I had to remove the commit that added a try-catch in the webworker code. The linter said "no".
- The last patch (v6.0.2) in Mochify adds Content-Type headers when serving files with the built-in https server. This fixes the webworker test case.

Build is green again. Coverage even went up a little. This is good to merge from my point of view.

**Comment by mantoni** [comment]
This fixes https://github.com/sinonjs/sinon/issues/1948

File: package.json (hunk 2 above)
Comment (mroderick) [reply package.json]: If it becomes a problem to hold off the stage-0 hordes, I'll take a hammer to the problem

**Comment by mroderick** [comment]
👍

**Comment by mantoni** [comment]
Thanks!

**Pull request merged**